sdks/python/apache_beam/io/gcp/bigquery_schema_tools.py

```
@@ -0,0 +1,115 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Tools used tool work with Schema types in the context of BigQuery.
+Classes, constants and functions in this file are experimental and have no
+backwards compatibility guarantees.
+NOTHING IN THIS FILE HAS BACKWARDS COMPATIBILITY GUARANTEES.
+"""
+import apache_beam as beam
+import numpy as np
+from typing import Optional
+from typing import Sequence
+
+# BigQuery types as listed in
+# https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types
+# with aliases (RECORD, BOOLEAN, FLOAT, INTEGER) as defined in
+# https://developers.google.com/resources/api-libraries/documentation/bigquery/v2/java/latest/com/google/api/services/bigquery/model/TableFieldSchema.html#setType-java.lang.String-
+BIG_QUERY_TO_PYTHON_TYPES = {
+    "STRING": "str",
+    "INTEGER": "np.int64",
+    "FLOAT64": "np.float64",
+    "BOOLEAN": "bool",
+    "BYTES": "bytes",
+    "TIMESTAMP": "apache_beam.utils.timestamp.Timestamp",
+    #TODO svetaksundhar@: Finish mappings for all BQ types
+}
+
+
+def produce_pcoll_with_schema(self, the_table_schema):
+  #type: [TableSchema] -> [UserType]
+
+  """Convert a schema of type TableSchema into a pcollection element.
+      Args:
+        the_table_schema: A BQ schema of type TableSchema
+      Returns:
+        usertype: type that can be used to work with pCollections.
+      """
+
+  the_schema = beam.io.gcp.bigquery_tools.get_dict_table_schema(
+      the_table_schema)
+  if the_schema == {}:
+    raise ValueError("The schema is empty")
+  i = 0
+  dict_of_tuples = []
+  for x in the_schema['fields']:
+    if the_schema['fields'][i]['type'] == 'STRING':
+      if the_schema['fields'][i]['mode'] == 'NULLABLE':
+        typ = Optional[str]
+      elif the_schema['fields'][i]['mode'] == 'REPEATED':
+        typ = Sequence[str]
+      else:
+        typ = str
```

nit: I'd make this `elif mode == ''` case, and the else case should raise an error. That way we won't accidentally accept a mode we don't understand.